### PR TITLE
feat/custom gh actions for reassure

### PR DESCRIPTION
**Pull request by ShaswatPrabhat**
Changes for [issue](https://github.com/callstack/reassure/issues/58)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -37,8 +37,21 @@ jobs:
...
-
-      - name: Run Danger.js
-        run: yarn danger ci
+          
+  build-validate-pref-tests-with-custom-action:
+    runs-on: ubuntu-latest
+    name: A job to test the new custom GH action
+    steps:
+      - uses: actions/checkout@v3
+        with:
```
Comment (mdjastrzebski) [inline .github/workflows/main.yml:46]: @ShaswatPrabhat Seems like using your ref is causing the related step to fail could you omit the `ref` options?

**Comment by mdjastrzebski** [comment]
@ShaswatPrabhat The PR is not ready for merging yet, but since you've already provided most of the setup I've awarded a `hacktoberfest-accepted` tag, so that it should count for your Hacktoberfest stats.

File: .github/workflows/main.yml (hunk 1 above)
Comment (ShaswatPrabhat) [reply .github/workflows/main.yml:46]: I had tried earlier with the same, let me try again @mdjastrzebski

Comment (ShaswatPrabhat) [reply .github/workflows/main.yml]: Got [this](https://github.com/callstack/reassure/actions/runs/3360152110/jobs/5568998534) now

Comment (ShaswatPrabhat) [reply .github/workflows/main.yml]: @mdjastrzebski might need some help here, whenever you have some time. Thanks!
